runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -187,30 +193,61 @@ private void registerIndices(final Descriptors.FieldDescriptor fieldDescriptor)
             if (fieldDescriptor.getType() == FieldDescriptor.Type.GROUP) {
                 throw new IllegalArgumentException("group is a deprecated, unsupported type");
             }
-            indices.put(indexName, getIndex(indexName, fieldDescriptor));
+            indices.put(indexName, getIndex(indexName, indexName, fieldDescriptor));
         } else if (fieldDescriptor.getOptions().getExtension(CorfuOptions.schema).hasNestedSecondaryKeys()) {
             if (fieldDescriptor.getType() == FieldDescriptor.Type.GROUP) {
                 throw new IllegalArgumentException("group is a deprecated, unsupported type");
             }
             // Parse and validate nested secondary key(s)
             String nestedSecondaryKeysString = fieldDescriptor.getOptions().getExtension(CorfuOptions.schema)
                     .getNestedSecondaryKeys();
-            List<String> secondaryKeys = parseAndValidateNestedSecondaryKeys(nestedSecondaryKeysString, fieldDescriptor);
-            for (String indexName : secondaryKeys) {
+            Map<String, String> secondaryKeysAliasToFullPath = parseAndValidateNestedSecondaryKeys(nestedSecondaryKeysString, fieldDescriptor);
+            for (Map.Entry<String, String> secondaryKey : secondaryKeysAliasToFullPath.entrySet()) {
                 // Place index name and a function on how the indexed value is computed
-                indices.put(indexName, getNestedIndex(indexName));
+                indices.put(secondaryKey.getValue(), getNestedIndex(secondaryKey.getValue(), secondaryKey.getKey()));
+                // For nested secondary indexes, aliases are supported (it can be user-defined or defaults to the last attributes name if not specified)
+                // e.g., full-path secondary key 'adult.children.child.age' defaults to alias 'age'
+                // We keep a map of alias to full path, as secondary indexes can ba accessed on either one of them
+                secondaryIndexAliasToFullPath.put(secondaryKey.getKey(), secondaryKey.getValue());
             }
         }
     }
 
+    private NestedSecondaryIndex parseSecondaryKey(String secondaryKey) {
```

Comment:
Are there other ways to define the alias instead of manually parsing the string, which seems not very neat and might be error prone? For example, can the `nested_secondary_keys` be a custom option, which is some structure, instead of a string? Or is it possible to define multiple field options, one is `nested_secondary_keys`, and another alias option?

Reply:
Sure, this sounds reasonable. 

I've made the change such that the new syntax is:

`message Adult {
    Person person = 1 [(org.corfudb.runtime.schema).nested_secondary_key = { index_path: "person.age" index_name: "age" },
                       (org.corfudb.runtime.schema).nested_secondary_key = { index_path: "person.children.child.age" index_name: "kidsAge" }];
    Company work = 2;
}`